bot/cogs/hiphopify.py

```diff
@@ -0,0 +1,184 @@
+
+log = logging.getLogger(__name__)
+
+
+class Hiphopify:
+    """
+    A set of commands to moderate terrible nicknames.
+    """
+
+    def __init__(self, bot: AutoShardedBot):
+        self.bot = bot
+        self.headers = {"X-API-KEY": SITE_API_KEY}
+
+    async def on_member_update(self, before, after):
+        """
+        This event will trigger when someone changes their name.
+        At this point we will look up the user in our database and check
+        whether they are allowed o change their names, or if they are in
+        hiphop-prison. If they are not allowed, we will change it back.
+        :return:
+        """
+
+        if before.display_name == after.display_name:
+            return  # User didn't change their nickname. Abort!
+
+        log.debug(
+            f"{before.display_name} is trying to change their nickname to {after.display_name}. "
+            f"Checking if the user is in hiphop-prison..."
+        )
+
+        response = await self.bot.http_session.get(
+            SITE_API_HIPHOPIFY_URL,
+            headers=self.headers,
+            params={"user_id": str(before.id)}
+        )
+
+        response = await response.json()
+
+        if response:
+
+            if after.display_name == response.get("forced_nick"):
+                return  # Nick change was triggered by this event. Ignore.
+
+            log.warning(
+                f"{after.display_name} is currently in hiphop-prison. "
+                f"Changing the nick back to {before.display_name}."
+            )
+            await after.edit(nick=response.get("forced_nick"))
+            await after.send(
+                "You have tried to change your nickname on the **Python Discord** server "
+                f"from **{before.display_name}** to **{after.display_name}**, but as you "
+                "are currently in hiphop-prison, you do not have permission to do so. "
+                "You will be allowed to change your nickname again at the following time:\n\n"
+                f"**{response.get('end_timestamp')}**."
+            )
+
+    @with_role(ADMIN_ROLE, OWNER_ROLE, MODERATOR_ROLE)
+    @command(name="hiphopify()", aliases=["hiphopify", "force_nick()", "force_nick"])
+    async def hiphopify(self, ctx: Context, member: Member, duration: str, forced_nick: str = None):
+        """
+        This command will force a random rapper name (like Lil' Wayne) to be the users
+        nickname for a specified duration. If a forced_nick is provided, it will use that instead.
+
+        :param ctx: Discord message context
+        :param ta:
+        If provided, this function shows data for that specific tag.
+        If not provided, this function shows the caller a list of all tags.
+        """
+
+        log.debug(
+            f"Attempting to hiphopify {member.display_name} for {duration}. "
+            f"forced_nick is set to {forced_nick}."
+        )
+
+        embed = Embed()
+        embed.colour = Colour.blurple()
```

Comment:
Any reason why this isn't in the creation of the embed, `Embed(colour=Colour.blurple())`?

Reply:
I suppose it could be. But we've done it this way in more or less all the other cogs, so it's done this way out of convention. We could change them all, but I'm not sure it's better because it seems to violate the Zen of Python somewhat. Flat is, after all, better than nested. Then again, I'm not sure it's that much worse, either. it's quite ¯\_(ツ)_/¯

Reply:
That's fine then